sources/main.cpp

```diff
@@ -91,16 +91,19 @@ int main(int argc, char* argv[]) {
     cmdLineHandler.process(app, appPath);
 
     // LOAD CONFIG
+    bool configError = false;
     if (!QFile::exists(cmdLineHandler.configFile())) {
         qFatal("App configuration file not found: %s", cmdLineHandler.configFile().toLatin1().constData());
         return 1;
     }
     Config* config = new Config(&engine, cmdLineHandler.configFile(), cmdLineHandler.configSchemaFile(), appPath);
     if (!config->isValid()) {
         qCCritical(CLASS_LC).noquote() << "Invalid configuration!" << endl << config->getError();
-        // TODO(marton) show error screen with shutdon / reboot / web-configurator option
+        configError = true;
     }
+
     engine.rootContext()->setContextProperty("config", config);
+    engine.rootContext()->setContextProperty("configError", configError);
```

Comment:
FYI: context properties will be deprecated in Qt6 according to this weeks Qt conference.
Ok for now, we'll need to refactor more stuff once we get there.

Reply:
Ok, I think we have some time until then :)